src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2575,13 +2597,14 @@ private TypeSymbol NominalSlotType(int slot)
         /// </summary>
         private void SetStateAndTrackForFinally(ref LocalState state, int slot, NullableFlowState newState)
         {
-            state[slot] = newState;
+            Debug.Assert(slot > 0);
+            SetState(ref state, slot, newState);
             if (newState != NullableFlowState.NotNull && NonMonotonicState.HasValue)
             {
                 var tryState = NonMonotonicState.Value;
                 if (tryState.HasVariable(slot))
                 {
-                    tryState[slot] = newState.Join(tryState[slot]);
+                    SetState(ref tryState, slot, newState.Join(GetState(ref tryState, slot)), useNotNullsAsDefault: true);
```

Comment:
:memo: I couldn't find or make a test where this matters, but I think it's correct given that the nullable walker's `NonMonotonicState` is initialized with `ReachableBottomState` (ie. explicitly filled with `NotNull` by default).

Reply:
I think `Join(state, NotNull)` is always equal to `state`. If `tryState` has no state for the given `slot`, it makes sense to me to just use `NotNull` as a default.